crates/loader/src/local/mod.rs

```diff
@@ -72,7 +72,11 @@ async fn prepare(
     let info = info(raw.info, &src);
 
     let mut config_root = raw.config.unwrap_or_default();
+    let component_ids: Vec<String> = raw.components.iter().map(|c| c.id.clone()).collect();
     for component in &mut raw.components {
+        if component_ids.contains(&component.id) {
+            panic!("cannot have duplicate component ID: {}", &component.id)
```

Comment:
Probably don't want to use a `panic!` here?

Reply:
We almost never want a library to be the source of a panic, which is why we try to return `Result` instead here.
(you can easily do that here with `bail!("<message>")`.

(we could do a much better job of returning actual typed errors instead, but for this PR, a `bail!` should be enough)